Menu redesign and slogans

janno42 opened the pull request:
I had some ideas for redesigning the navbar and footer. This is in preparation for #1042 but also to give them a sleeker design in general. The navbar is now smaller and the footer is not fixed anymore (the navbar remains fixed on top).
Note that the results link will also become a single link without dropdown in the future. Non-staff users will then just have the user dropdown for the appropriate choices of settings/rewards/logout in the upper right corner.
Many individual things differ from the current implementation and I'm looking forward to a discussion and even more ideas.

Here are some screenshots:
![login](https://user-images.githubusercontent.com/1781719/35970039-1bb29316-0cc9-11e8-88a4-0126ca813fef.PNG)
![student](https://user-images.githubusercontent.com/1781719/35970049-1feb51ca-0cc9-11e8-95dd-20b1406655f5.PNG)
![responsible](https://user-images.githubusercontent.com/1781719/35970050-21453d38-0cc9-11e8-9653-906fce2ecc0a.PNG)
![res_top](https://user-images.githubusercontent.com/1781719/35970053-245db2d4-0cc9-11e8-9710-82eb94deabfd.PNG)
![res_bottom](https://user-images.githubusercontent.com/1781719/35970054-25b85300-0cc9-11e8-90d7-ee9eb96c75ca.PNG)

He3lixxx commented:
I really like it 👍 

One thing we may want to think about: The menu items in the header are grey on dark-grey. The backronym is white on dark-grey so it's a bit more eye-catching. This doesn't really decrease usability, but we may want to think about whether we want more focus on the slogan than on the menu. Maybe we want to increase the contrast on the menu items a bit and then make the menu items and the slogan the same color?

janno42 commented:
I added slogans and refactored the navbar and footer.
Now ready for review.

janno42 commented:
One test fails because it asserts a certain body of a page, but the slogan is randomly selected on each page load so the message might differ. Any suggestions for a good test solution?

karyon commented:
you could try `random.seed(42)` at the beginning of the test.

janno42 commented:
that's nice, thanks.

karyon commented:
on a second look, that test seems to be entirely broken. it visits `/results/semester/<id>`, which does not have the public view parameter, `/results/semester/<id>/course/<id>` does. could you fix that please?

janno42 commented:
check

karyon commented:
* the text in the navbar is not vertically centered for me, there is more padding to the top (screenshot below)
* i like the language buttons. they seem to be correctly centered, however, the text inside them is not again (in this zoom level by one pixel only though, in other zoom levels by two)
* the hamburger menu would benefit from some vertical padding i think.
* the problems/feedback uses a smaller font, which looks a bit weird (i realize the button would be quite large otherwise, but still)
* actually, by now i think the navbar is too small. i'm used to rather large navbars from most sites, also it looks a bit old-fashioned. maybe there's a middle ground?
* i would prefer to have the rewards in the navbar and not a dropdown to make it more visible.


![image](https://user-images.githubusercontent.com/1891915/36330975-3f1586d4-136c-11e8-9bbc-99293bd0c8f6.png)

janno42 commented:
note to self: tooltips are broken

janno42 commented:
- navbar height will be set to 42px
- center content will stay centered between the right and left content so that we don't lose too much space (sorry @He3lixxx)
- vertical centering stuff will hopefully be done
- text size in feedback button will be increased
- language change buttons will move into user dropdown
- rewards will move to main navbar
- currently either tooltips are positioned correctly *or* the footer stays at the bottom for pages with content smaller than one screen. this will be fixed but is one of the most annoying things possible so i just wrote done what needs to be done and will wait until i actually have enough motivation to do that.

janno42 commented:
note to self: think about an integration of the breadcrumbs into the navbar.

karyon commented:
dump from the chat:
* removing the breadcrumb altogether is still weird. e.g. grade publishers have no link to the course when uploading a document for the course, except for the back button.
* we'll need to reevaluate caching of the navbar if there's changing stuff in it
* staff users could have an additional "current semester" in the navbar. (there's more room for that with #1109!)
* or a click on "manage" in the navbar leads to the current semester, while the other stuff is accessible through a smaller dropdown-symbol beneath that
* a second navbar below the first one for staff users was proposed

janno42 commented:
- done
  - height was increased
  - feedback button is larger
  - rewards are in navbar
  - tooltips are correctly positioned
- won't fix
  - i didn't move the language buttons into the user dropdown - i don't like hiding them there because users might want to change the language per course during evaluation
  - vertical alignment: this seems to be a general issue of text positioning in firefox. it usually displays vertically centered text lower than every other browser (see images below). if it's really important we can add some browser-specific settings but i would prefer not having to.
    - chrome: ![chrome](https://user-images.githubusercontent.com/1781719/37873963-a37fdcba-3026-11e8-9af7-ac11eaf3b3e7.png)
    - firefox: ![firefox](https://user-images.githubusercontent.com/1781719/37873964-a7299a0e-3026-11e8-91c6-ddac8c044318.png)
- todos:
  - correct collapsing
  - breadcrumbs?

janno42 commented:
i now added a suggestion for breadcrumbs. the breadcrumbs bar is integrated into the navbar and fixed to the top.

on staff and grades views breadcrumbs work as before (only without the leading "EvaP", because that linked to the same page as the large logo right above the breadcrumbs does).
the course edit page for contributors, the course vote page for students and the course result pages for all users now also show the course name (and semester) on the breadcrumbs bar.
breadcrumbs have been removed from all other pages because they were unnecessary.

![breadcrumbs](https://user-images.githubusercontent.com/1781719/37911015-3b3b3f5c-310f-11e8-8672-60f6075150f7.PNG)
![breadcrumbs2](https://user-images.githubusercontent.com/1781719/37911016-3b599614-310f-11e8-98d0-87bc18fa9e20.PNG)

janno42 commented:
collapsing was also changed. ready for review.

karyon commented:
very nice!

![image](https://user-images.githubusercontent.com/1891915/37920685-db828aac-3127-11e8-9455-4c40b465380f.png)

on smaller screens, there is a bit too little spacing between the breadcrumb and the headline. if there is any reasonably easy way around that, please do that, otherwise i think we can live with that.

karyon commented on evap/evaluation/templates/base.html:
```diff
@@ -37,20 +37,23 @@
+    {% cache 3600 navbar request.session.session_key LANGUAGE_CODE %}
+        {% include_navbar user LANGUAGE_CODE %}
+    {% endcache %}
+    {% if user.is_staff or user.is_grade_publisher %}
```
reviewers don't get the breadcrumb, is that still correct?

janno42 replied:
this is outdated :)

karyon commented on evap/grades/templates/grades_base.html:
```diff
@@ -4,9 +4,9 @@
-        <li class="breadcrumb-item"><a href="{% url 'grades:index' %}">{% trans 'Grade publishing' %}</a></li>
-    {% else %}
+    {% if disable_breadcrumb_grades or not user.is_grade_publisher %}
         <li class="breadcrumb-item">{% trans 'Grade publishing' %}</li>
```
why is this displayed even if `not user.is_grade_publisher`?

janno42 replied:
for staff users who can access the page from the staff semester view

karyon commented on evap/staff/templates/staff_base.html:
```diff
@@ -2,11 +2,17 @@
+        <ul class="breadcrumb">
+            {% block breadcrumb %}
+                {% if disable_breadcrumb_staff or not user.is_staff %}
+                    <li class="breadcrumb-item">{% trans 'Manage' %}</li>
```
same here, when is this shown to non-staff users?

janno42 replied:
for reviewers :)

karyon commented on evap/results/templates/results_course_detail.html:
```diff
@@ -1,14 +1,8 @@
-    <li class="breadcrumb-item"><a href="{% url 'results:semester_detail' course.semester.id %}">{{ course.semester.name }}</a></li>
-    <li class="breadcrumb-item">{{ course.name }}</li>
-{% endblock %}
+{% block title %}{% trans 'Results' %} - {{ course.name }} - {{ course.semester.name }} - {{ block.super }}{% endblock %}
```
order of this stuff as discussed

janno42 commented on evap/grades/templates/grades_base.html:
```diff
@@ -2,11 +2,17 @@
+                    <li class="breadcrumb-item">{% trans 'Grade publishing' %}</li>
+                {% else %}
+                    <li class="breadcrumb-item"><a href="{% url 'grades:index' %}">{% trans 'Grade publishing' %}</a></li>
+                {% endif %}
```
we decided to keep it
